docs/docs/mdx/migrate-remark-to-mdx.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Migrating Remark to MDX
+---
+
+For people who already have an existing blog using gatsby-transformer-remark, but want to use MDX, you can swap out the remark transformer plugin with gatsby-plugin-mdx and touch little code otherwise.
+
+## Prerequisites
+
+- An existing site that builds pages using `gatsby-transformer-remark` ([gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog) will be used as a reference in this guide).
+
+## Adding in gatsby-plugin-mdx
+
+Add gatsby-plugin-mdx (and its peer dependencies) to your package.json and remove gatsby-transformer-remark.
+
+```shell
+npm install @mdx-js/mdx @mdx-js/react gatsby-plugin-mdx
+npm remove gatsby-transformer-remark
+```
+
+## Replacing gatsby-transformer-remark with gatsby-plugin-mdx
+
+In your gatsby-config.js file, replace gatsby-transformer-remark with gatsby-plugin-mdx. All sub-plugins of gatsby-transformer-remark can still work with gatsby-plugin-mdx by updating the plugins option to gatsbyRemarkPlugins.
+
+```diff:title=gatsby-config.js
+{
+- resolve: `gatsby-transformer-remark`
++ resolve: `gatsby-plugin-mdx`
+  options: {
+-   plugins: [
++   gatsbyRemarkPlugins: [
+```
+
+## Update gatsby-node.js
+
+In the `createPages` API call, when you query for `allMarkdownRemark`, replace it with `allMdx`.
+
+```diff:title=gatsby-node.js
+const result = await graphql(
+  `
+    {
+-     allMarkdownRemark(
++     allMdx(
+        sort: { fields: [frontmatter___date], order: DESC }
+        limit: 1000
+      ) {
+```
+
+Also, update `onCreateNode` which creates the blog post slugs to watch for the node type of `Mdx` instead of `MarkdownRemark`.
+
+```diff:title=gatsby-node.js
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+
+-  if (node.internal.type === `MarkdownRemark`) {
++  if (node.internal.type === `Mdx`) {
+```
+
+## Update File Extension
+
+Where your markdown files live, change the file extension to `.mdx`.
+
+Also, you can tell `gatsby-plugin-mdx` to accept both `md` and `mdx` files by adding the `extensions` option in your gatsby-config entry.
```

Comment:
I might move this up to the section changing gatsby-config stuff and make the extension recommendation the default. Then note that they also have the option of changing the file extensions directly.

Reply:
I moved this section up as a subsection of "Replacing gatsby-transformer-remark with gatsby-plugin-mdx" and cleaned it up a little based on @marcysutton's suggestions.